scripts/build.mjs

```diff
@@ -317,25 +317,34 @@ async function buildCssBundles(buildFn, themes, assets, mainCssFile = null) {
         mainCssFile = path.join(cssSrcDir, "main.css");
     }
     const bundleCss = await buildFn(mainCssFile);
+    const themesRelPath = 'themes';
+    const themesRoot = path.join(cssSrcDir, themesRelPath);
     await assets.write(`hydrogen.css`, bundleCss);
     for (const theme of themes) {
         const themeRelPath = `themes/${theme}/`;
         const themeRoot = path.join(cssSrcDir, themeRelPath);
         const assetUrlMapper = ({absolutePath}) => {
-            if (!absolutePath.startsWith(themeRoot)) {
-                throw new Error("resource is out of theme directory: " + absolutePath);
-            }
-            const relPath = absolutePath.substr(themeRoot.length);
+            const relPath = assetRelPath(absolutePath, themeRoot, themesRoot);
             const hashedDstPath = assets.resolve(path.join(themeRelPath, relPath));
             if (hashedDstPath) {
-                return hashedDstPath.substr(themeRelPath.length);
+                return assetRelPath(hashedDstPath, themeRelPath, themesRelPath);
             }
         };
         const themeCss = await buildFn(path.join(themeRoot, `theme.css`), assetUrlMapper);
         await assets.write(path.join(themeRelPath, `bundle.css`), themeCss);
     }
 }
 
+function assetRelPath(assetPath, themeRoot, themesRoot) {
+    if (assetPath.startsWith(themeRoot)) {
+        return assetPath.substr(themeRoot.length);
+    } else if (assetPath.startsWith(themesRoot)) {
+        // Themes are allowed to access assets in other themes
```

Comment:
Note to self to check what the output of this actually looks like when running the build

Reply:
Here's an example of the target build. The `element-dark` theme accesses the fonts from the `element` theme via a relative path.

![Screenshot 2021-05-18 at 21 20 07](https://user-images.githubusercontent.com/1137962/118710643-db238580-b81e-11eb-9943-7225018fcefd.png)